src/SDL_shadercross.c

```diff
@@ -1656,6 +1656,68 @@ static SPIRVTranspileContext *SDL_ShaderCross_INTERNAL_TranspileFromSPIRV(
     return transpileContext;
 }
 
+void SDL_ShaderCross_GetIOVars(
+    spvc_compiler compiler,
+    spvc_reflected_resource* reflected_resources,
+    size_t num_vars,
+    SDL_ShaderCross_IOVarMetadata* vars
+) {
+    Uint32 offset = 0;
+    for (size_t i = 0; i < num_vars; i++) {
+        SDL_ShaderCross_IOVarMetadata* var = &vars[i];
+        spvc_reflected_resource* resource = &reflected_resources[i];
+        spvc_type type = spvc_compiler_get_type_handle(compiler, resource->base_type_id);
+
+        switch (spvc_type_get_basetype(type)) {
+            case SPVC_BASETYPE_INT8:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_BYTE;
+                break;
+            case SPVC_BASETYPE_UINT8:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_UBYTE;
+                break;
+            case SPVC_BASETYPE_INT16:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_SHORT;
+                break;
+            case SPVC_BASETYPE_UINT16:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_USHORT;
+                break;
+            case SPVC_BASETYPE_INT32:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_INT;
+                break;
+            case SPVC_BASETYPE_UINT32:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_UINT;
+                break;
+            case SPVC_BASETYPE_INT64:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_LONG;
+                break;
+            case SPVC_BASETYPE_UINT64:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_ULONG;
+                break;
+            case SPVC_BASETYPE_FP16:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_HALF;
+                break;
+            case SPVC_BASETYPE_FP32:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_FLOAT;
+                break;
+            case SPVC_BASETYPE_FP64:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_DOUBLE;
+                break;
+            default:
+                var->vector_type = SDL_SHADERCROSS_IOVAR_TYPE_UNKNOWN;
+                break;
+        }
+
+        Uint32 vector_size = spvc_type_get_vector_size(type);
+        var->vector_size = vector_size;
+        size_t name_length = SDL_min(SDL_utf8strlen(resource->name) + 1, SDL_SHADERCROSS_MAX_IO_VAR_NAME);
```

Comment:
Perhaps we should emit a warning when the variable name is too long?

Reply:
> Great! @thatcosmonaut Would it make sense to add a `-WX`-like mode to the cli util? In general, developers dislike warnings.

Probably a good idea.